Packs/TrendMicroEmailSecurity/Integrations/TrendMicroEmailSecurityEventCollector/TrendMicroEmailSecurityEventCollector.py

```diff
@@ -0,0 +1,380 @@
+    event_type: str,
+    hide_sensitive: bool,
+) -> tuple[list[dict], str | None]:
+    """
+    Fetch the event logs by type.
+    The fetch loop continues as long as it does not encounter one of the following:
+        - The size of the event list is equal to the limit
+        - Calling the api returned No content
+        - The nextToken did not return from the api call
+
+    Args:
+        client (Client): The client for api calls.
+        start (str), end (str): Start and end time period to retrieve logs.
+        limit (int): Maximum number of log items to return.
+        token (str | None): Token for retrieve the next set of log items.
+        event_type (str): The type of event log to return.
+
+    Returns:
+        tuple[list[dict], str | None]: List of the logs returned from trend micro,
+                                       The token that returned for the next fetch.
+    """
+    if token:
+        # The unquoting  is must for the api call
+        token = urllib.parse.unquote(token)
+
+    params = assign_params(
+        start=start,
+        end=end,
+        token=token,
+        type=event_type,
+    )
+
+    next_token: str | None = None
+    events_res: list[dict] = []
+    while len(events_res) < limit:
+        params["limit"] = min(limit - len(events_res), 500)
+
+        try:
+            res = client.get_logs_request(event_type, params)
+        except NoContentException:
+            next_token = None
+            demisto.debug(f"No content returned from api, {params=}")
+            break
+
+        if res.get("logs"):
+            # Iterate over each event log, update their `type` and `_time` fields
+            for event in res.get("logs"):
+                event.update({"_time": event.get("timestamp"), "logType": event_type})
+                if hide_sensitive:
+                    remove_sensitive_from_events(event)
+
+            events_res.extend(res.get("logs"))
+        else:
+            next_token = None
+            break
+
+        if next_token := res.get("nextToken"):
+            params["token"] = urllib.parse.unquote(next_token)
+        else:
+            next_token = None
+            demisto.debug(f"No returned `nextToken` for the {event_type} type")
+            break
+
+    demisto.debug(
+        f"The fetch process of the type {event_type} has ended,"
+        f" the amount of logs for this type: {len(events_res)}"
+    )
+    return events_res, next_token
+
+
+def main() -> None:  # pragma: no cover
+    params = demisto.params()
+    args = demisto.args()
+
+    base_url = params["url"].strip("/")
+    username = params.get("credentials")["identifier"]
+    api_key = params.get("credentials")["password"]
+    verify = argToBoolean(params.get("verify", "false"))
+    proxy = argToBoolean(params.get("proxy", "false"))
+    first_fetch = order_first_fetch(params.get("first_fetch") or "3 days")
+
+    should_push_events = argToBoolean(args.get("should_push_events", False))
```

Comment:
this is a required argument, no need for `.get`
```suggestion
    should_push_events = argToBoolean(args["should_push_events"])
```

Reply:
When the `test-module` command and the `fetch-events` command are run they do not have the `should_push_events` argument